vtex-apps/order-items
# Add better error handling for addItems and updateItems mutations

#### What problem is this solving?

Currently, when a error occurs in the `addToCart` or `updateItems` mutations, the error is ignored and the item will still be in the user's cart, instead of being removed (or added back). This PR aims to fix this.

[**Quick review link**](https://github.com/vtex-apps/order-items/pull/22/files?diff=unified&w=1)

[Related clubhouse story](https://app.clubhouse.io/vtex/story/35714/remo%C3%A7%C3%A3o-de-item-exibi%C3%A7%C3%A3o-de-mensagem-no-carrinho-quando-graphql-retorna-erro).

#### How should this be manually tested?

I prepared [this workspace](https://addfail--checkoutio.myvtex.com/cart) in which you can use to test these changes. You can use the "add random items" in Cartman to add the items. This workspace contains a linked version of `checkout-graphql` that will throw 50% of the time. Just try to add/update some items and see that if something goes wrong, the cart is updated accordingly.

#### Checklist/Reminders


#### Type of changes


✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements
